examples/vulkan/Vulkan.cpp

```diff
@@ -1838,16 +1838,16 @@ class VulkanExample
     void setupTextureImage()
     {
         // Load the image data
-        sf::Image imageData;
+        std::optional<sf::Image> imageData = sf::Image::loadFromFile("resources/logo.png");
 
-        if (!imageData.loadFromFile("resources/logo.png"))
+        if (!imageData.has_value())
         {
             vulkanAvailable = false;
             return;
         }
 
```

Comment:
I'd love if we used the pattern you proposed in discord where you create a new `Image` and move the contents of the optional into it after the nullness of the optional has been checked and handled. Then you can rename the existing `imageData` to `maybeImageData`. I like how the `maybe` prefix is very `grep`able and will easily show up on code reviews should someone try to use this optional after it's been moved from or tries to assign to the optional again.

I think it's a big help if we reduce this optional back down to a value ASAP so that we're not using a type that does more than we want. In the spirit of your CppCon talk, lets use the least powerful tool possible and values are certainly less powerful than optionals.

Reply:
This is a very good point. 💯 

The code path that _statically_ knows that an optional is non-empty, should express this with the type system.

I'm fine with the "maybe" prefix. It would be possible to do hacks with local closures that do scoped initialization, but this is more confusing than a simple variable. Unfortunately in C++ it's neither possible to late-init variables (with static analysis) nor to have expression blocks for initialization values. Maybe there's another pattern to shadow the original variable.